src/simple_lottery.rs

```diff
@@ -0,0 +1,185 @@
+//! ALBA Telescope with simple lottery construction given in Section 4.1.
+
+/// Setup output parameters
+#[derive(Debug, Clone)]
+pub struct LotterySetup {
+    /// Proof size (in Sp elements)
+    pub u: u64,
+    /// Lottery oracle probability
+    pub p: f64,
+}
+
+impl LotterySetup {
+    /// Setup algorithm taking as input the security parameters, the set size,
+    /// and the lower bound and returns LotterySetup.
+    pub fn new(lambda_sec: f64, lambda_rel: f64, n_p: u64, n_f: u64) -> Self {
+        let (u, mu) = Self::compute_u_mu(lambda_sec, lambda_rel, n_p, n_f);
+
+        Self {
+            u,
+            p: mu as f64 / n_p as f64,
+        }
+    }
+
+    /// Setup algorithm taking the security parameters, the set size, the lower
+    /// bound and the expected number of participants and returns LotterySetup
+    /// if the parameters are in bounds, None otherwise.
+    pub fn from(lambda_sec: f64, lambda_rel: f64, n_p: u64, n_f: u64, mu: u64) -> Option<Self> {
+        let p = mu as f64 / n_p as f64;
+        let u_opt = Self::compute_u(lambda_sec, lambda_rel, n_p, n_f, p);
+        u_opt.map(|u| Self { u, p })
+    }
+
+    // Setup algorithm taking as input the security parameters, the set size,
+    /// and the lower bound and returns the minimum number of participants
```

Comment:
I am confused about this function - minimal over what? Also, `mu` in the paper is defined as some expected value. What does `minimal number of participants` mean - minimal over what?